odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,497 @@
+
+    The template ``I`` is deformed by a displacement field ``v`` as:
+
+        (I, v) --> I(Id + v)
+
+    Parameters
+    ----------
+    template : `DiscreteLpVector`
+        Template to be deformed by a displacement field.
+    displacement : `ProductSpaceVector`
+        The vector field (displacement field) used in the linearized
+        deformation.
+
+    Returns
+    -------
+    deformed_template : `numpy.ndarray`
+        Deformed template as an numpy array.
+
+    Examples
+    --------
+    Deform a 1D function ``template`` that has value ``1`` at 3:rd sample
+    point using a displacement field ``displacement_field`` that has
+    value ``-0.2`` in the 4:th sample point. The outcome should be an
+    array with values ``1`` at 3:rd and 4:th positions.
+
+    >>> import odl
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+    >>> template = space.element([0, 0, 1, 0, 0])
+    >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+    >>> linear_deform(template, displacement_field)
+    array([ 0.,  0.,  1.,  1.,  0.])
+    """
+    assert template in displacement[0].space
+
+    image_pts = template.space.grid.points()
+    image_pts += np.asarray(displacement).T
+
+    return template.interpolation(image_pts.T, bounds_check=False)
+
+
+def deform_grad(grad_f, displacement):
+    """Compute the deformation of the template gradient.
+
+    The template gradient ``grad(I)`` after deformation with displacement
+    field ``v``: grad(I)(Id + v)
+
+    Parameters
+    ----------
+    grad_f: `ProductSpaceVector`
+        Gradient of the template, i.e. a vector field on the
+        image domain
+    displacement : `ProductSpaceVector`
+        The vector field (displacement field) used in the linearized
+        deformation.
+    Examples
+    --------
+    >>> import odl
+
+    Define ``gradOp`` as the gradient operator, let it act on ``template``.
+
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> template = space.element([0, 0, 1, 0, 0])
+    >>> gradOp = odl.Gradient(template.space, method='forward')
+    >>> template_grad = gradOp(template)
+    >>> template_grad
+    ProductSpace(uniform_discr(0.0, 1.0, 5), 1).element([
+        [0.0, 5.0, -5.0, 0.0, 0.0]
+    ])
+
+    Define a displacement field ``displacement_field``.
+
+    >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+    >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+
+    Calculate linearized deformation of the ``template_grad`` (gradient of
+    the template) using the displacement field ``displacement_field``.
+
+    >>> deform_grad(template_grad, displacement_field)
+    ProductSpace(uniform_discr(0.0, 1.0, 5), 1).element([
+        [0.0, 5.0, -5.0, -5.0, 0.0]
+    ])
+    """
+    temp = [linear_deform(gf, displacement) for gf in grad_f]
+    return displacement.space.element(temp)
+
+
+class LinDeforFixedTempOp(Operator):
```

Comment:
Like the abbreviation pattern. Slight suggestion: `Defor -> Deform` - much more clarity with 1 additional letter; also `Template` should be spelt out, otherwise it's hard to know what `Temp` means (temporary?).